source/Cosmos.System2/Audio/AudioMixer.cs

```diff
@@ -0,0 +1,177 @@
+    /// </summary>
+    /// <remarks>
+    /// The <see cref="AudioMixer"/> class does not implement any kind of compressor or
+    /// limiter, and as such, the resulting audio may introduce hard-clipping.
+    /// </remarks>
+    public class AudioMixer : AudioStream {
+        uint pos = 0;
+        uint sampleRate;
+        AudioBuffer mixBuffer;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AudioMixer"/> class,
+        /// with the specified initial internal mixing buffer size and sample format.
+        /// </summary>
+        /// <remarks>
+        /// This constructor should be used when the sizes and formats of the
+        /// buffers that will be given to the mixer are known. If there is a
+        /// mis-match between the mixing format and/or the buffer size and the given buffer,
+        /// the internal mixing buffer will be re-initialized with the right values.
+        /// </remarks>
+        /// <param name="format">The resulting format of the mixing process.</param>
+        /// <param name="bufferSize">The size of the mixing buffer, in audio samples.</param>
+        public AudioMixer(SampleFormat format, int bufferSize)
+        {
+            mixBuffer = new AudioBuffer(bufferSize, format);
+            streamReader = new AudioBufferReader(mixBuffer);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AudioMixer"/> class.
+        /// </summary>
+        /// <remarks>
+        /// The internal mixing buffer will be set to <see langword="null"/> and
+        /// will be initialized with the correct format and buffer size as soon
+        /// as a read operation will be attempted.
+        /// </remarks>
+        public AudioMixer()
+        {
+            mixBuffer = null;
+            streamReader = null;
+        }
+
+        /// <summary>
+        /// The list of streams that the audio mixer should process.
+        /// This list will be dynamically modified as streams get depleted.
+        /// </summary>
+        public List<AudioStream> Streams { get; private set; } = new List<AudioStream>();
+
+        /// <summary>
+        /// The size of the internal mixing buffer, in audio samples.
+        /// </summary>
+        public int BufferSize => mixBuffer.Size;
+
+        #region Standard AudioStream Members
+        public override uint SampleRate {
+            get => sampleRate;
+            set => sampleRate = value;
+        }
+
+        public override bool Depleted => false;
+
+        /// <summary>
+        /// The position, in samples, of the mixer audio stream. Attempting to set this
+        /// property to another value will result in an <see cref="InvalidOperationException"/>.
+        /// </summary>
+        public uint Position
+        {
+            get => pos;
+            set => throw new InvalidOperationException("Cannot set the position of an AudioMixer.");
```

Comment:
Why not just remove the setter?

Reply:
Cannot remove the setter in an abstract class, unfortunately :(

Reply:
I thought `AudioMixer` only inherits from `AudioStream` which doesnt have `Position`. Only `SeekableAudioStream` has a `Position`.

Reply:
Oops, yeah, you're right! I was thinking of an old version where all AudioStreams had that property, sorry ^^'>